src/core_ocean/shared/mpas_ocn_gm.F

```diff
@@ -486,86 +488,93 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, diagnosticsPool, &
-            gradZMidTopOfEdge(maxLevelEdgeTop(iEdge)+1,iEdge) = gradZMidEdge(maxLevelEdgeTop(iEdge),iEdge)
-         end if
-      end do
-      !$omp end do
+               ! Approximation of values on the top and bottom interfaces through the idea of having ghost cells 
+               ! above the top and below the bottom layers of the same depths and density.
+               gradDensityTopOfEdge(1,iEdge) = gradDensityEdge(1,iEdge)
+               gradDensityTopOfEdge(maxLevelEdgeTop(iEdge)+1,iEdge) = gradDensityEdge(maxLevelEdgeTop(iEdge),iEdge)
+               gradZMidTopOfEdge(1,iEdge) = gradZMidEdge(1,iEdge)
+               gradZMidTopOfEdge(maxLevelEdgeTop(iEdge)+1,iEdge) = gradZMidEdge(maxLevelEdgeTop(iEdge),iEdge)
+            end if
+         end do
+         !$omp end do
 
-      !$omp do schedule(runtime) private(k)
-      do iEdge = 1, nEdges
-         if (maxLevelEdgeTop(iEdge) .GE. 1) then
-            do k = 1, maxLevelEdgeTop(iEdge)+1
-               gradDensityConstZTopOfEdge(k,iEdge) = gradDensityTopOfEdge(k,iEdge) - dDensityDzTopOfEdge(k,iEdge) &
+         !$omp do schedule(runtime) private(k)
+         do iEdge = 1, nEdges
+            if (maxLevelEdgeTop(iEdge) .GE. 1) then
+               do k = 1, maxLevelEdgeTop(iEdge)+1
+                  gradDensityConstZTopOfEdge(k,iEdge) = gradDensityTopOfEdge(k,iEdge) - dDensityDzTopOfEdge(k,iEdge) &
                                                    * gradZMidTopOfEdge(k,iEdge)
-            end do
-         end if
-      end do
-      !$omp end do
-      !$omp end parallel
+               end do
+            end if
+         end do
+         !$omp end do
+         !$omp end parallel
 
+         nEdges = nEdgesArray(3)
          ! For config_GM_closure = 'N2_dependent' use a scaling to taper gmBolusKappa
          ! based on stratification relative to the maximum in the column
          if (local_config_GM_kappa_lat_depth_variable) then
             !$omp parallel
             !$omp do schedule(runtime) private(k, BruntVaisalaFreqTopEdge, maxN)
-            do iCell=1,nCells
+            do iEdge=1,nEdges
                k=1
-               maxN = max(BruntVaisalaFreqTop(k, iCell), 0.0_RKIND)
+               cell1 = cellsOnEdge(1,iEdge)
+               cell2 = cellsOnEdge(2,iEdge)
+               BruntVaisalaFreqTopEdge = 0.5_RKIND*(BruntVaisalaFreqTop(k,cell1) + &
+                                                    BruntVaisalaFreqTop(k,cell2))
+               maxN = max(BruntVaisalaFreqTopEdge, 0.0_RKIND)
                do k = 2, maxLevelCell(iCell)
-                 maxN = max(maxN,max(BruntVaisalaFreqTop(k, iCell), 0.0_RKIND))
+                 BruntVaisalaFreqTopEdge = 0.5_RKIND*(BruntVaisalaFreqTop(k,cell1) + &
+                                                      BruntVaisalaFreqTop(k,cell2))
+                 maxN = max(maxN,max(BruntVaisalaFreqTopEdge, 0.0_RKIND))
                enddo
 
                do k = 1,maxLevelCell(iCell)
```

Comment:
This is `maxLevelCell(iCell)` as well.  For my own reference, would `maxLevelCell(cell1)` equal `maxLevelCell(cell2)` in all cases?

Reply:
I think they should all be `maxLevelEdgeTop(iEdge)`.  cell1 and cell2 would, in general, have different values of `maxLevelCell`.  That is why there are two arrays for the edge: `maxLevelEdgeTop` and `maxLevelEdgeBottom`.